src/env/mod.rs

```diff
@@ -2,14 +2,35 @@ use crate::api::customization::Customization;
 use crate::api::firmware_protection::FirmwareProtection;
 use crate::api::upgrade_storage::UpgradeStorage;
 use crate::ctap::status_code::Ctap2StatusCode;
-use crate::ctap::Channel;
+use crate::ctap::{Channel, Transport};
 use persistent_store::{Storage, Store};
 use rng256::Rng256;
 
+#[cfg(feature = "std")]
+pub mod host;
 #[cfg(feature = "std")]
 pub mod test;
 pub mod tock;
 
+pub enum SendOrRecvStatus {
+    // FIXME: Replace Option<SendOrRecvStatus>::None with SendOrRecvStatus::TimedOut?
+    Error,
+    Sent,
+    Received(Transport),
+}
+
+pub trait IOChannel {
+    // FIXME: Separate type for timeout.
+    fn recv_with_timeout(&mut self, buf: &mut [u8; 64], timeout: isize)
+        -> Option<SendOrRecvStatus>;
+    fn send_or_recv_with_timeout(
+        &mut self,
+        buf: &mut [u8; 64],
```

Comment:
I inferred that you want all traffic for your Host env to go through [CTAPHID](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#usb-commands). Other transports would have different packet sizes. But since there is only one channel, it seems you want to reuse it.